content/documentation/pages/5-feature-guides/1-streams/3-partitioning.md

```diff
@@ -4,6 +4,138 @@ title: 'Data Partitioning'
 description: 'How to use partitioning'
 ---
 
-Partitioning support allows for content-based routing of payloads to downstream application instances in an event streaming pipeline.
-This is especially useful when you want to have your downstream application instances processing data from specific partitions.
+Partitioning is a critical concept in stateful processing, for either performance or consistency reasons, to ensure that all related data is processed together.
+For example, in a time-windowed average calculation example, it is important that all measurements from any given sensor are processed by the same application instance.
+Alternatively, you may want to cache some data related to the incoming events so that it can be enriched without making a remote procedure call to retrieve the related data.
+
+Partitioning support allows for `content` based routing of payloads to the `downstream` application instances in a streaming data pipeline.
```

Comment:
It'd be good to remove backticks from content and downstream - they are not code fragments